src/modules/keyboardmanager/dll/KeyboardEventHandlers.cpp

```diff
@@ -611,10 +624,27 @@ namespace KeyboardEventHandlers
         LPINPUT keyEventList = new INPUT[size_t(key_count)]();
         memset(keyEventList, 0, sizeof(keyEventList));
 
-        // Use the shortcut flag to ensure these are not intercepted by any remapped keys or shortcuts
+        // Use the suppress flag to ensure these are not intercepted by any remapped keys or shortcuts
         KeyboardManagerHelper::SetKeyEvent(keyEventList, 0, INPUT_KEYBOARD, VK_NUMLOCK, KEYEVENTF_KEYUP, KeyboardManagerConstants::KEYBOARDMANAGER_SUPPRESS_FLAG);
         KeyboardManagerHelper::SetKeyEvent(keyEventList, 1, INPUT_KEYBOARD, VK_NUMLOCK, 0, KeyboardManagerConstants::KEYBOARDMANAGER_SUPPRESS_FLAG);
         UINT res = ii.SendVirtualInput((UINT)key_count, keyEventList, sizeof(INPUT));
         delete[] keyEventList;
     }
+
+    // Function to ensure Ctrl/Shift/Alt modifier key state is not detected as pressed down by applications which detect keys at a lower level than hooks when it is remapped
+    void ResetIfModifierKeyForLowerLevelKeyHandlers(InputInterface& ii, DWORD key)
+    {
+        // If the argument is either of the Ctrl/Shift/Alt modifier key codes
+        if (KeyboardManagerHelper::IsModifierKey(key) && !(key == VK_LWIN || key == VK_RWIN || key == CommonSharedConstants::VK_WIN_BOTH))
+        {
+            int key_count = 1;
+            LPINPUT keyEventList = new INPUT[size_t(key_count)]();
+            memset(keyEventList, 0, sizeof(keyEventList));
+
+            // Use the suppress flag to ensure these are not intercepted by any remapped keys or shortcuts
+            KeyboardManagerHelper::SetKeyEvent(keyEventList, 0, INPUT_KEYBOARD, (WORD)key, KEYEVENTF_KEYUP, KeyboardManagerConstants::KEYBOARDMANAGER_SUPPRESS_FLAG);
+            UINT res = ii.SendVirtualInput((UINT)key_count, keyEventList, sizeof(INPUT));
+            delete[] keyEventList;
+        }
+    }
```

Comment:
@arjunbalgovind Can you add Unit tests for the two cases that you are adding.  I'm imagining something like: 

1.  HandleSingleKeyRemapEvent_ShouldSendVirutalInputWithSuppressFlag_WhenCtrlAltShiftIsMappedToCapsLock
2. HandleSingleKeyRemapEvent_ShouldSendVirutalInputWithSuppressFlag_WhenCapsLockIsMappedToCtrlAltShift

And assert that SendVirtualInput is only called once each with the expected suppress flag.

Reply:
Added the tests as you suggested. I only tested for Caps->Ctrl and Ctrl->Caps because doing it for each modifier would be redundant.
Since the Microsoft C++ Native Unit Testing Framework doesn't have an API for asserting the number of times a method is called, I added a count variable to the SendVirtualInput method. I made it generic such that you can set the condition under which you would want that to be incremented so that the class is still generic enough to be usable by other PowerToys.